File: src/types/tag.ts
export type TagScope = "movie" | "actor" | "both";

export interface Tag {
  id: number;
  name: string;
  description: string | null;
  scope: TagScope;
}

export interface TagGroup {
  id: number;
  name: string;
  description: string | null;
  sort_order: number;
}

export interface Genre {
  id: number;
  name: string;
  description: string | null;
}

export interface TagGroupWithTags extends TagGroup {
  tags: Tag[];
}

File: src-tauri/src/models/tag.rs
use serde::{Deserialize, Serialize};

#[allow(dead_code)]
#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Tag {
    pub id: i64,
    pub name: String,
    pub description: Option<String>,
    pub scope: String,
}

#[allow(dead_code)]
#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct TagGroup {
    pub id: i64,
    pub name: String,
    pub description: Option<String>,
    pub sort_order: i32,
}

#[allow(dead_code)]
#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Genre {
    pub id: i64,
    pub name: String,
    pub description: Option<String>,
}

#[allow(dead_code)]
#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct TagGroupWithTags {
    pub id: i64,
    pub name: String,
    pub description: Option<String>,
    pub sort_order: i32,
    pub tags: Vec<Tag>,
}
